### PR TITLE
[Merged by Bors] - refactor: improvements to lint_style

### PR DESCRIPTION
**Pull request by grunweg**
To prepare for #14273; pulled out for easy of reviewing.
Commits can be reviewed individually.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Tactic/Linter/TextBased.lean [hunk 1]
```diff
@@ -128,7 +127,8 @@ instance : BEq ErrorContext where
       && (ctx.error).normalise == (ctx'.error).normalise
 
 /-- Output the formatted error message, containing its context.
-`style` specifies if the error should be formatted for humans or for github output matchers -/
+`style` specifies if the error should be formatted for humans to read, github problem matchers
+to consume or for the style exceptions file. -/
```
Comment (semorrison) [inline Mathlib/Tactic/Linter/TextBased.lean:131]:
```suggestion
to consume, or for the style exceptions file. -/
```

File: Mathlib/Tactic/Linter/TextBased.lean [hunk 2]
```diff
@@ -194,7 +200,8 @@ def parseStyleExceptions (lines : Array String) : Array ErrorContext := Id.run d
   Array.filterMap (parse?_errorContext ·) (lines.filter (fun line ↦ !line.startsWith "--"))
 
 /-- Print information about all errors encountered to standard output.
-`style` specifies if we print errors for humand or github consumption. -/
+`style` specifies if the error should be formatted for humans to read, github problem matchers
+to consume or for the style exceptions file. -/
```
Comment (semorrison) [inline Mathlib/Tactic/Linter/TextBased.lean:204]:
```suggestion
to consume, or for the style exceptions file. -/
```

**Comment by semorrison** [comment]
bors merge

**Comment by grunweg** [comment]
Thank you for the fast review!

**Pull request closed without merging**